### ISSUE TITLE
What if the user doesn't select from the suggestions list and types it out? Is there any possible way to change plaintext to markup.

### ISSUE THREAD

**Issue by RamakrishnaChilaka** [signavio/react-mentions#347]

**Comment by kashifshamaz21** [comment]
@RamakrishnaChilaka No, currently there is no way to achieve what you're seeking. If user doesn't select from the list of suggestions, then its hard to know the intent of the typed string. And moreover, the user's selection from the list triggers a lot of other functions/ callbacks, bookeeping etc. So I doubt if this feature can be supported as such. That's just my 2 cents.
@jfschwarz and others might have better insights on this though.

**Comment by Badbreaddead** [comment]
I also need it...

**Comment by Lozovoi-Rodion** [comment]
That is will be really useful.

**Comment by xtrinch** [comment]
Implementing this in <b>our</b> code base would be hacking around the custom markup your library provides (e.g. @\[xtrinch\](id:215)) and the string user types, (e.g. @xtrinch). It really does not seem like a very good option and I think this belongs inside the library.

Also in the case when extracting mentions from textarea value (https://github.com/signavio/react-mentions/issues/319) is just absolutely necessary, the manually typed strings will not be accounted for at all.

**Comment by byronwall** [comment]
I also thought this would be a deal breaker, but I settled on a quick work around.  I changed the `markup` property to just be `@__id__@` which means I can get the goodness of this library just by ending my "exact" entry with the trigger character.  For my app, that's a workable solution.  The nice thing is that the `onChange` handler properly detects these items even when they're not in the `data` list.  That might actually be a separate bug, but it makes my life easy. For me this was desired because I want to be able to "create" new mentions automatically (I'm simply tagging tasks in a to-do app).  Obviously there's extra code to process the new items, but the core idea is simple.

```jsx
<Mention
              markup="@__id__@"
              trigger="@"
              data={names}
              style={defaultMentionStyle}
            />
```
